python/lsstimport.py

```diff
@@ -37,59 +40,62 @@
 # resolved by setting dynamical library loading flags.
 RTLD_GLOBAL = -1
 RTLD_NOW = -1
-try:
-    import dl
-    if hasattr(dl, 'RTLD_GLOBAL'):  RTLD_GLOBAL = dl.RTLD_GLOBAL
-    if hasattr(dl, 'RTLD_NOW'):     RTLD_NOW    = dl.RTLD_NOW
-except ImportError:
-    # 64bit linux does not have a dl module...
-    pass
-except SystemError:
-    # ...if it does it should throw a SystemError
-    pass
 
-try:
-    if RTLD_GLOBAL < 0:
-        import lsst64defs
-        RTLD_GLOBAL = lsst64defs.RTLD_GLOBAL   # usually 0x00100
-    if RTLD_NOW < 0:
-        import lsst64defs
-        RTLD_NOW = lsst64defs.RTLD_NOW         # usually 0x00002
-    DLFLAGS = RTLD_GLOBAL|RTLD_NOW
-except ImportError:
-    sys.stderr.write(
-        "Could not import lsst64defs; please ensure the base package has been built (not just setup).\n"
-    )
+# For portability we try a number of different options
+# Unfortunately on a Mac running 2.7 RTLD_NOW is not available anywhere
+options = ('os', 'DLFCN', 'ctypes', 'dl', 'lsst64defs')
+for mod in options:
```

Comment:
My understanding is that this happens once on first import of the `lsstimport` module. Currently the `import dl` always fails and `lsst64defs` is always loaded. With this patch `os` will be loaded and won't be useful unless you are running python3 (but is a common module that is likely to be needed later) and then on Linux `DLFCN` will be loaded which will then work. On a Mac running Python 2.x `RTLD_NOW` will always fail to load until the last possible moment.

I could put in prior knowledge and do a version check for python3 and always load `lsst64defs` (which probably has a bad name at that point as it's not 64-bit specific anything) in every other case. We have to create `lsst64defs.py` regardless as we do not know at build time the python variant that we will be using at run time (if we take as a given that scons may not be running the same python).

This all means that we could drop all the self-determination stuff and always ask `lsst64defs.py` for the answer....